src/Assert.php

```diff
@@ -1688,6 +1855,17 @@ protected static function strlen($value)
         return \mb_strlen($value, $encoding);
     }
 
+    private static function throwException($message, $exception = null)
+    {
+        if (!is_null($exception)) {
+            if ($exception instanceof Exception) {
+                throw $exception;
```

Comment:
Seems like you don't actually need a message in this case, so you can use the only argument as message/exception and do not touch literally every method in the lib :wink:

Reply:
That's a great point... this could be a mixed type, if string then call reportInvalidArgument, if exception just throw it. Thanks

Reply:
Here is something I came up with but there is no way to pass the 3 arguments for the Exception class
$message = "", $code = 0, Throwable $previous = null

Using the below approach there is no way to add a code or a custom message to the exception instance.

```php
<?php

namespace Webmozart\Assert\Tests;

use Exception;
use Webmozart\Assert\Assert as WebmozartAssert;

class CustomException extends Exception
{}

class Assert extends WebmozartAssert
{
    static function reportInvalidArgument($arg)
    {
        if (class_exists($arg)) {
            throw new $arg();
        }
        parent::reportInvalidArgument($arg);
    }
}


class CustomExceptionTest extends BaseTestCase
{
    public function testThings()
    {
        $this->setExpectedException(CustomException::class);
        Assert::true(false, CustomException::class);
    }
}
```